src/App.tsx

```diff
@@ -1,24 +1,215 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useEffect,
+  useMemo,
+  useState,
+  useCallback,
+} from 'react';
 import { UserWarning } from './UserWarning';
+import { TodoList } from './components/TodoList';
+import { TodoFilter } from './components/TodoFilter';
+import { Header } from './components/Header';
+import {
+  getTodos,
+  createTodo,
+  removeTodo,
+  updateTodo,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { ErrorNotification } from './components/ErrorNotification';
+import { Filters } from './types/filters';
 
-const USER_ID = 0;
+const USER_ID = 10953;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [err, setErr] = useState<string | null>(null);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [loadingTodos, setLoadingTodos] = useState([0]);
+  const [filter, setFilter] = useState(Filters.ALL);
+  const activeTodos = todos.filter(todo => !todo.completed);
+  const completedTodos = todos.filter(todo => todo.completed);
```

Comment:
you now create these variables on each render, even if you don't need them
Let's just calculate it if you need it